### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by jacob-berger**

### CODE REVIEW

**Comment by mfirmin1** [comment]
I will review this.

**Comment by mfirmin1** [comment]
1) This was done correctly.
2) Conditional Nullability was done correctly. 
3) Classes were created with the correct properties, and certain properties were indeed readonly as specified.
4) Non-default constructors were added properly.
5) Unit test were added, checking for nullable types in the correct properties and checking if it would gift and user were successfully created when proper input is used.
6) Directory.Build.Props was added properly but Directory.Build.Targets was not added.
7) IntellitectAnalayzer was added properly.
8) There is not a global suppression file.
9) It seems like it was properly implemented, but I don't know Azure Dev ops well enough to be certain.
10) This seems to have been done properly.
If you have any questions let me know.

File: SecretSanta/src/SecretSanta.BusinessTests/GiftTests.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            Assert.IsNotNull(gift.User);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void NewGift_NullTitle_ThrowsException()
+        {
+            //Arrange
+            //Act
+            Gift gift = new Gift(1, null!, "Description", "URL", new User(2, "First", "Last", new List<Gift>()));
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.BusinessTests/GiftTests.cs]: For cases where you don't care about the return value you can tell the compiler to ignore the value.
```suggestion
            _ = new Gift(1, null!, "Description", "URL", new User(2, "First", "Last", new List<Gift>()));
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+- master
+- Assignment*
+
+pool:
+  vmImage: 'windows-latest'
+
+jobs:
+  - job: ParallelTesting
+    strategy:
+      parallel: 5
```
Comment (Keboo) [inline azure-pipelines.yml]: This is close. This sets up the pipeline to run things in parallel, but still requires splitting the up across each of these 5 parallel jobs.

**Pull request closed without merging**